test/volumes-list-sizes.test.js

```diff
@@ -0,0 +1,86 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+/*
+ * Copyright (c) 2017, Joyent, Inc.
+ */
+
+var assert = require('assert-plus');
+var test = require('tape').test;
+
+var common = require('./common');
+var mod_config = require('../lib/config.js');
+var mod_testConfig = require('./lib/config');
+
+var CONFIG = mod_config.configure();
+
+if (CONFIG.experimental_nfs_shared_volumes !== true) {
+    console.log('experimental_nfs_shared_volumes setting not enabled, ' +
+        'skipping tests');
+    process.exitCode = 0;
+} else {
+    var CLIENTS;
+    var CLIENT;
+    var SERVER;
+
+    var testVolumeName = common.createResourceName('test-volumes-basic');
+    var testVolume;
+    var testVolumeStorageVmUuid;
+
+    test('setup', function (t) {
+        common.setup({clientApiVersion: '~8.0'}, function (_, clients, server) {
+            CLIENTS = clients;
+            CLIENT = clients.user;
+            SERVER = server;
+
+            t.end();
+        });
+    });
+
+    /*
+     * This is necessary so that we proceed with the rest of the tests suite
+     * only after the entry for the newly added user (including its default
+     * fabric network used to provision volumes) is present in UFDS.
+     */
+    test('getting config from ufds', function (t) {
+        mod_testConfig.waitForAccountConfigReady(CLIENT,
+            function onConfigReady(configReadyErr) {
+                t.ifErr(configReadyErr, 'newly created user\'s config should ' +
+                    'eventually be created');
+                t.end();
+            });
+    });
+
+    test('should be able to get volume sizes',
+        function (t) {
+            CLIENT.get('/my/volumes/sizes?type=tritonnfs',
```

Comment:
Thanks for pushing me to add a test here on this, doing so exposed a pretty annoying bug. Because of the ordering of the routes, we were calling GetVolume instead of ListVolumeSizes since the URL:

```
/my/volumes/sizes
```

was being treated as a volume named 'sizes'. That said, I wonder if this is actually a more fundamental problem here. Since this means there cannot be a volume named 'sizes'. I'll fix things as they are now, but talk to you about this before pushing anything further.